api/src/main/java/org/openmrs/Condition.java

```diff
@@ -264,7 +264,7 @@ public void setOnsetDate(Date onsetDate) {
 	 * @return endDate - a date object that shows the end date of the condition
 	 */
 	public Date getEndDate() {
-		return endDate;
+		return endDate != null ? (Date) endDate.clone() : new Date();
```

Comment:
Same as above.

Reply:
Again [this](https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java#L50) statement here

Reply:
As said above